source/community.applications/usr/local/emhttp/plugins/community.applications/skins/Narrow/skin_helpers.php

```
@@ -0,0 +1,1651 @@
+      }
+    } elseif (! ($template['Blacklist'] ?? false) || ! ($template['Compatible'] ?? false)) {
+      if ($template['InstallPath']) {
+        $userTemplate = readXmlFile($template['InstallPath'], false, false);
+        if (! $template['Blacklist']) {
+          $actionsContext[] = ["icon" => "ca_fa-install", "text" => tr("Reinstall"), "action" => "popupInstallXML('".addslashes($template['InstallPath'])."','user','','".portsUsed($userTemplate)."');"];
+          $actionsContext[] = ["divider" => true];
+        }
+        $actionsContext[] = ["icon" => "ca_fa-delete", "text" => tr("Remove from Previous Apps"), "alternate" => tr("Remove"), "action" => "removeApp('{$template['InstallPath']}','{$template['Name']}');"];
+      } else {
+        if (! ($template['BranchID'] ?? null)) {
+          if (is_file(CA_PATHS['dockerManTemplates']."/my-{$template['Name']}.xml")) {
```

PHPStan (2.1.46)

[error] Constant CA_PATHS not found.
Learn more at https://phpstan.org/user-guide/discovering-symbols

(constant.notFound)